File: Algorithms/BinarySearch.java
import java.util.Scanner;

public class BinarySearch {

    // Binary Search method
    public static int binarySearch(int[] arr, int target) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;

            // Check if target is present at mid
            if (arr[mid] == target) {
                return mid;
            }

            // If target greater, ignore left half
            if (arr[mid] < target) {
                left = mid + 1;
            } 
            // If target is smaller, ignore right half
            else {
                right = mid - 1;
            }
        }

        // Target not found in the array
        return -1;
    }

    public static void main(String[] args) {
        int[] array = { 3, 4, 5, 6, 7, 8, 9 };
        int n = array.length;

        // Get input from user for element to be searched
        Scanner input = new Scanner(System.in);
        System.out.println("Enter element to be searched:");
        int element = input.nextInt();
        input.close();

        // Call the binary search method
        int result = binarySearch(array, element);

        // Print result based on return value of binarySearch
        if (result == -1) {
            System.out.println("Element not found");
        } else {
            System.out.println("Element found at index " + result);
        }
    }
}

File: Binarysearch.py
//Python code for binary search algorithm
def binary_search(arr, target):
    # Set initial left and right pointers
    left, right = 0, len(arr) - 1

    while left <= right:
        # Find the middle index
        mid = left + (right - left) // 2
        
        # Check if the target is at the middle
        if arr[mid] == target:
            return mid
        # If target is greater, ignore the left half
        elif arr[mid] < target:
            left = mid + 1
        # If target is smaller, ignore the right half
        else:
            right = mid - 1

    # Target not found in the array
    return -1

# Example usage
arr = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]
target = 7
result = binary_search(arr, target)

if result != -1:
    print(f"Target found at index: {result}")
else:
    print("Target not found")
